metapool/sample_sheet.py

```diff
@@ -1667,6 +1668,19 @@ def make_sample_sheet(metadata, table, sequencer, lanes, strict=True):
     messages = sheet._validate_sample_sheet_metadata(metadata)
 
     if len(messages) == 0:
+        # if the user did not *explicitly* set the strict value
+        if strict is None:
+            # TODO: this is a temporary measure. Katharoseq-enabled sample
+            #  sheets may or may *not* have katharoseq controls, and if they
+            #  don't, they fail during strict adding, even though they're legal
+            # NB: the below is duck-typing.  It isn't checking whether the
```

Comment:
I don't understand this TODO. I think it's fine to fail if the Katharoseq sample-sheet does not have Katharoseq samples. If we allow this, what prevents users to simply define everything as Katharoseq sample-sheet?

Reply:
Hmm, I'm confused here!  What I did in this edit was just make sure that the katharoseq functionality actually acts the way it is written to say it acts--I didn't define its intended actions myself ;)  @antgonza, looks like you approved a PR in April (https://github.com/biocore/metagenomics_pooling_notebook/pull/193) that implements *optional* katharoseq columns, so I was under the impression that maybe our end goal **was** to make every sample sheet katharoseq-capable (whether or not we actually used katharoseq for that particular experiment).  I thought this was analogous to wanting to make every sample sheet going forward multiproject-capable, whether or not we ended up putting multiple projects on the plates for in any given experiment.

Do we NOT want this?  If not, then I agree that we should make more cases fail; it will be a fairly key change to the way that katharoseq support is currently envisioned in the code, though.

Reply:
I see your point and maybe the way to go in the future: allow users to define whatever they need based on the samples (well, sample names) and not by the "type". If you agree with this, I'm OK with it.

Reply:
If I understand what you are saying, I think that it doesn't require changes to the current katharoseq functioning ... do I have that right?

Reply:
I think so too.

Reply:
Right. Since katharoseq controls were presented as an extension of BLANKs, what defines a katharoseq-enabled sample-sheet was previously the presence of sample-names beginning with 'katharo' and now their presence in the new section. It's true you still need to use one of the newer sample-sheet types in order to get the katharoseq functionality, but this is because we want to preserve the versions that we processed past sheets with.